core-api/src/main/java/org/silverpeas/core/calendar/Recurrence.java

```diff
@@ -366,20 +446,78 @@ public int hashCode() {
         .toHashCode();
   }
 
-  protected Recurrence() {
+  @Override
+  public Recurrence clone() {
+    try {
+      Recurrence clone = (Recurrence) super.clone();
+      clone.id = null;
+      clone.daysOfWeek = new HashSet<>(daysOfWeek);
+      clone.exceptionDates = new HashSet<>(exceptionDates);
+      return clone;
+    } catch (CloneNotSupportedException e) {
+      throw new SilverpeasRuntimeException(e);
+    }
   }
 
-  @PrePersist
-  protected void generateId() {
-    this.id = UUID.randomUUID().toString();
+  /**
+   * Is this recurrence identical in value than the specified one.
+   * @param recurrence the recurrence with which this recurrence is compared to.
+   * @return true if this recurrence is same as the given one, false otherwise.
+   */
+  boolean sameAs(final Recurrence recurrence) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "sameAs" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)